libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -308,6 +312,26 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 			")\n";
 		break;
 	}
+	case FunctionType::Kind::Assert:
+	case FunctionType::Kind::Require:
+	{
+		solAssert(arguments.size() > 0, "Expected at least one parameter for require/assert");
+
+		for (auto& arg: arguments)
+			arg->accept(*this);
+
+		auto requireOrAssertFunction = m_utils.requireOrAssertFunction(
+			functionType->kind() == FunctionType::Kind::Assert,
+			arguments.size() > 1 ? arguments[1]->annotation().type : nullptr
+		);
+
+		m_code << requireOrAssertFunction << "(" << m_context.variable(*arguments[0]);
+		if (arguments.size() > 1)
```

Comment:
Could assert that `size() <= 2`

Reply:
True. I imitated the assert I found in the EVM implementation assuming the original author had their reason for doing it that way.